### ISSUE TITLE
Node 21 error: `Cannot read properties of undefined (reading 'pop')`

### ISSUE THREAD

**Issue by DawidJanczak** [sass/dart-sass#2127]
Hello,

We're getting the above mentioned error after upgrading Node to 21. We're using webpack and `sass-loader` if this changes anything. Here's the stacktrace:
```
ERROR in ./src/styles/ui_components/_tabs.scss (./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[7].use[1]!./node_modules/sass-loader/dist/cjs.js??ruleSet[1].rules[7].use[2]!./src/styles/ui_components/_tabs.scss)
Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
TypeError: Cannot read properties of undefined (reading 'pop')
    at Object.<anonymous> (/var/app/node_modules/sass/sass.node.js:2:43)
    at Module._compile (node:internal/modules/cjs/loader:1369:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1427:10)
    at Module.load (node:internal/modules/cjs/loader:1201:32)
    at Module._load (node:internal/modules/cjs/loader:1017:12)
    at Module.require (node:internal/modules/cjs/loader:1229:19)
    at require (node:internal/modules/helpers:177:18)
    at getDefaultSassImplementation (/var/app/node_modules/sass-loader/dist/utils.js:37:10)
    at getSassImplementation (/var/app/node_modules/sass-loader/dist/utils.js:46:30)
    at Object.loader (/var/app/node_modules/sass-loader/dist/index.js:23:55)
 @ ./src/styles/ui_components/_tabs.scss 8:6-208 22:17-24 26:7-21 52:25-39 53:36-47 53:50-64 57:6-67:7 58:54-65 58:68-82 64:42-53 64:56-70 66:21-28 77:0-178 77:0-178 78:22-29 78:33-47 78:50-64 55:4-68:5
 @ ./src/components/tabs.ts 2:0-44
 @ ./src/entry_points/interproject.ts 5:0-28
```

Reported line looks like this `const library = globalThis._cliPkgExports.pop();` so it looks like `globalThis._cliPkgExports` is undefined.

**Comment by euaaaio** [comment]
Seems to be related to dart-lang/sdk#53784

**Comment by DawidJanczak** [comment]
I was thinking about that, but all occurrences of that other error mention `indexOf` and there's no references to `navigator` in _this_ code. But still could be that, I'll definitely re-test it when that fix is in :)

**Comment by euaaaio** [comment]
Yeah. That's why I mentioned them here. 
It scares me more that they haven't released a fix in 5 days.

**Comment by gajus** [comment]
Same here. Vite fails with:

```
  [WebServer] Preprocessor dependency "sass" failed to load:
  Cannot read properties of undefined (reading 'indexOf')
  [WebServer] Preprocessor dependency "sass" failed to load:
  Cannot read properties of undefined (reading 'pop')
  [WebServer] 1:18:33 PM [vite] Internal server error: Preprocessor dependency "sass" failed to load:
```

Pasting in the error so Vite folks discover the thread when Google-ing it.

**Comment by nex3** [comment]
Can someone provide a minimal reproduction for this? I've tried manually editing `sass.dart.js` to work around https://github.com/dart-lang/sdk/issues/53784, but when I do so Dart Sass loads and runs fine in Node 21 for me (via both ESM and CJS loads).

**Comment by sannajammeh** [comment]
Getting this as well in the latest Next.js 13.5.6 pages directory

**Comment by VishnuJin** [comment]
facing the same with Vite, any workarounds ?

**Comment by sannajammeh** [comment]
> facing the same with Vite, any workarounds other than downgrade of nodejs?

Unfortunately its the only work around

**Comment by sebmaz93** [comment]
upgrading to node v21.1.0 fixed the issue for me

**Comment by nex3** [comment]
Since this seems to be fixed in Node 21.1, I'm closing it out.
